x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -411,9 +458,9 @@ export class Session {
 
   /**
    * Creates logger scoped to a specified session ID.
-   * @param sid Session ID to create logger for.
+   * @param [sid] Session ID to create logger for.
    */
-  private getLoggerForSID(sid: string) {
-    return this.options.logger.get(sid?.slice(-10));
+  private getLoggerForSID(sid?: string) {
+    return this.options.logger.get(sid?.slice(-10) ?? 'x'.repeat(10));
```

Comment:
__note:__ alternatively we can just use parent "context" if SID isn't provided.

Reply:
Yep, I like `no_session` and it's exactly 10 chars, thanks! :slightly_smiling_face: